### PR TITLE
Support absolute-form and authority-form URLs by web server

### LINKED ISSUE #6227: aiohttp 3.8.0 wrong value of Request.rel_url.raw_path
### Describe the bug

Wrong value of Request.rel_url.raw_path when request was made with url in absolute form

### To Reproduce

1. Make a simple server app with logging hook
```
from aiohttp import web


async def log(request, response):
    print('rel_url:', request.rel_url)
    print('raw_path:', request.rel_url.raw_path)


async def hello(request):
    return web.Response(text="Hello, world")


app = web.Application()
app.add_routes([web.get('/test', hello)])
app.on_response_prepare.append(log)
web.run_app(app, host='127.0.0.1', port=3322)
```
2. Make a request using relative form of url:
`echo -e "GET /test HTTP/1.1\r\n\r\n" | nc -N 127.0.0.1 3322`
3. Make a request using absolute form of url:
`echo -e "GET http://aaa.com/test HTTP/1.1\r\n\r\n" | nc -N 127.0.0.1 3322`

### Expected behavior

Server logs the "raw_path" as `/test` in both cases

### Logs/tracebacks

aiohttp==3.8.0
```python-traceback
rel_url: /test
raw_path: /test

rel_url: http://abc.com/test
raw_path: http://abc.com/test
```

aiohttp==3.7.4
```python-traceback

rel_url: /test
raw_path: /test

rel_url: http://abc.com/test
raw_path: /test
```

### Python Version

```console
Python 3.8.7
```

### aiohttp Version

```console
aiohttp==3.8.0
```

### multidict Version

```console
multidict==5.2.0
```

### yarl Version

```console
yarl==1.7.2
```

### OS

Ubuntu 18.04

### Related component

Server

### Additional context

_No response_

### Code of Conduct

- [x] I agree to follow the aio-libs Code of Conduct

### PR DESCRIPTION
**Pull request by asvetlov**
Fixes #6227

### CODE REVIEW

**Comment by asvetlov** [comment]
@webknjaz I think many proxy tests were literally wrong.
`reqest.path_qs` is for path with query_string and fragment parts, not for absolute URL

**Comment by webknjaz** [comment]
> @webknjaz I think many proxy tests were literally wrong.

Yeah, I noticed some oddities but haven't looked into it closely besides making what was failing pass. One of the things I've filed an issue for, as a result of that, was https://github.com/aio-libs/aiohttp/issues/6031 — have you seen it? I think it's something that you could have better context about than me.
FWIW I think it'd be more reliable to have the tests rewritten as end-to-end using the proxy.py fixture that is now implemented in the tests rather than hoping that the mocks are correct.

File: CHANGES/6227.bugfix [hunk 1]
```diff
@@ -0,0 +1 @@
+Supported authority-form and absolute-form URLs when sent to the server.
```
Comment (webknjaz) [inline CHANGES/6227.bugfix]:
```suggestion
Started supporting ``authority-form`` and ``absolute-form`` URLs on the server-side.
```

**Comment by asvetlov** [comment]
Rewriting tests with proxy.py is a great idea. Sorry, I have no resources for performing it myself.
Regarding mocking in affected tests: `test_utils.py` is not touched. I mean the real request should have very similar behavior to created by `make_mocked_request()` factory.

**Comment by asvetlov** [comment]
Backported by 17ba1f0f and 535c58ec

**Pull request merged**